web/src/Sidebar.tsx

```diff
@@ -20,6 +22,7 @@ class SidebarItem {
   isTiltfile: boolean
   status: ResourceStatus
   hasEndpoints: boolean
+  lastBuildDur: Date | null
```

Comment:
Not sure if this is _the way_, but I tend to think of it case by case. Build Duration is a thing that we optionally show, only when there exists a last build with a successfully completed duration. So I would expect it to be undefined in lots of cases: before a given resource has ever been updated (like on initial startup), or maybe when a last build was aborted (with manual trigger).

So, I think Last Build Duration can be null in these cases where we decide we don't want to show it. (Like, we should tell you how much time elapsed because you canceled the last build, but what's the point of that?)